text/metrics/0113-exemplars.md

```diff
@@ -0,0 +1,94 @@
+# Integrate Exemplars with Metrics
+
+This OTEP adds exemplar support to aggregations defined in the Metrics SDK.
+
+## Definition
+
+Exemplars are example data points for aggregated data. They provide specific context to otherwise general aggregations. For histogram-type metrics, exemplars are points associated with each bucket in the histogram giving an example of what was aggregated into the bucket. Exemplars are augmented beyond just measurements with references to the sampled trace where the measurement was recorded and labels that were attached to the measurement.
+
+## Motivation
+
+Defining exemplar behaviour for aggregations allows OpenTelemetry to support exemplars in Google Cloud Monitoring.
+
+Exemplars provide a link between metrics and traces. Consider a user using a Histogram aggregation to track response latencies over time for a high QPS server. The histogram is composed of buckets based on the speed of the request, for example, "there were 55 requests that took 400-500 milliseconds". The user wants to troubleshoot slow requests, so they would need to find a trace where the latency was high. With exemplars, the user is able to get an exemplar trace from a high latency bucket, an exemplar trace from a low latency bucket, and compare them to figure out the reason for the high latency.
+
+Exemplars are meaningful for all aggregations where relevant traces can provide more context to the aggregation, as well as when exemplars can display specific information not otherwise shown in the aggregation (for example, the full set of labels where they otherwise might be aggregated away).
+
+## Internal details
+
+An exemplar is defined as:
+
+```
+message Exemplar {
+  // Numerical value of the measurement that was recorded. Only one of these two fields is
+  // used for the data, depending on its type
+  double double_value = 0;
+  int64 int64_value = 1;
+  
+  // Exact time that the measurement was recorded
+  fixed64 time_unix_nano = 2;
+
+  // 'label:value' map of all labels that were provided by the user recording the measurement
+  repeated opentelemetry.proto.common.v1.StringKeyValue labels = 3;
```

Comment:
Should we consider add a field to indicate the original Metrics instrument type (one of six)?

Reply:
I assume that parameter would belong on the Aggregator instead, since Exemplars won't leave the scope of Aggregator and each aggregator has a single associated metric instrument (although I could be mistaken)

Reply:
We have all the metric instrument info in SDK side. but If we'll store all the raw metric events in Exemplars and pass it to Collector. I may want to keep the same info in Exemplar and Collector will be able to retrieve it.

Reply:
fair point, it just feels like a lot of unnecessary space for an attribute that will be the same for every exemplar in the list. If the instrument type was attached to the aggregator, the exporter could grab it and send it separately alongside the exemplars, which to me makes more sense. (there is no case where an aggregator is a raw list of exemplars, it will have an aggregator object wrapping it that can have the instrument property. so the information should always get to at least the exporter)

Reply:
The instrument type should be part of the metric descriptor, but the exemplars are associated with aggregations. We intend to support derived aggregations in the protocol, to support the Views API. I agree that we should not include parts of the instrument descriptor in the exemplar, but then how should we tie the output of an aggregation to the descriptor of the instrument that was used? I believe this is a question for the Views API design.